unit4/finalExercise.py

```diff
@@ -0,0 +1,190 @@
+from typing import Generator
+
+SEC_IN_MIN: int = 60
+MIN_IN_HOUR: int = 60
+HOUR_IN_DAY: int = 24
+FEB_IN_LEAP_YEAR: int = 29
+MONTHS_IN_YEAR: int = 12
+FEB: int = 2
+DAYS_IN_MONTH: list[int] = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+
+
+def gen_secs() -> Generator:
+    """
+    Generates seconds in a minute
+    :return: a generator containing the second
+    """
+    sec: int = 0
+    while sec < SEC_IN_MIN:
+        yield sec
+        sec += 1
+
+
+def gen_minutes() -> Generator:
+    """
+    Generates minutes in an hour
+    :return: a generator containing the minute
+    """
+    mnts: int = 0
+    while mnts < MIN_IN_HOUR:
+        yield mnts
+        mnts += 1
+
+
+def gen_hours() -> Generator:
+    """
+    Generates hours in a day
+    :return: a generator containing the hour
+    """
+    hour: int = 0
+    while hour < HOUR_IN_DAY:
+        yield hour
+        hour += 1
+
+
+def gen_time() -> Generator:
+    """
+    Generates the time in a day
+    :return: a generator containing a string that represents the time
+    """
+    end_of_clock: bool = False
+    sec_gen: Generator[int] = gen_secs()
+    mnts_gen: Generator[int] = gen_minutes()
+    hours_gen: Generator[int] = gen_hours()
+
+    sec: int = sec_gen.__next__()
```

Comment:
```suggestion
    sec: int = next(sec_gen)
```
as previously mentioned, make sure to change all the occurrences of `__next__()`